caikit/core/module_type.py

```diff
@@ -238,12 +239,16 @@ def decorator(cls_):
                         )
                     cls_.TASK_CLASS = parent_task
                 else:
-                    if not task:
-                        # TODO: raise once @tasks are integrated into service generation
-                        # raise TypeError(f"task argument is required for @{cls}")
-                        pass
                     cls_.TASK_CLASS = task
 
+                # Parse the `train` and `run` signatures
+                cls_.RUN_SIGNATURE = CaikitMethodSignature(cls_, "run")
+                cls_.TRAIN_SIGNATURE = CaikitMethodSignature(cls_, "train")
```

Comment:
this is 🌶️ ! I'm assuming we can use these signatures to send exact parameter types when making `run` or `train` calls through the servicers?

Reply:
We sure can!

And I still think the servicers should get a copy of the `ServicePackage` that they're serving anyway to have more 🌶️🌶️🌶️ info about what to do. It could be the case that somebody passes a model id to the incorrect task, and the servicer won't know that if it doesn't have a handle to the caikit rpc object that it's supposed to be serving, to see if it aligns with what the module says its signature is